examples/audio/melgan_spectrogram_inversion.py

```diff
@@ -0,0 +1,548 @@
+"""
+"""
+## Introduction
+
+Autoregressive vocoders have been ubiquitous for a majority of the speech processing history
+but they lack parallelism. [MelGAN](https://arxiv.org/pdf/1910.06711v3.pdf) is a
+non-autoregressive, fully convolutional vocoder architecture used for purposes ranging
+from spectral inversion and speech enhancement to speech synthesis when used as a decoder
+with models like Tacotron2 or FastSpeech.
+
+In this tutorial, we will have a look at the MelGAN architecture and how it can achieve
+fast spectral inversion, i.e. conversion of spectrograms to audio waves. The MelGAN
+implemented in this tutorial is similar to the original implementation with only the
+difference of method of padding for convolutions where we will use 'same' instead of
+reflect padding.
+"""
+
+"""
+## Imports
+"""
+
+"""shell
+!pip install -qqq tensorflow_addons
+!pip install -qqq tensorflow-io
+"""
+
+import tensorflow as tf
+import tensorflow_io as tfio
+import numpy as np
+import librosa.display
+import glob
+from tensorflow.keras.layers import *
+from tensorflow_addons.layers import WeightNormalization
+from sklearn.model_selection import train_test_split
+
+# Defining hyperparameters
+
+DESIRED_SAMPLES = 8192
+TEST_SPLIT = 0.2
+LEARNING_RATE_GEN = 1e-5
+LEARNING_RATE_DISC = 1e-6
+BATCH_SIZE = 16
+
+mse = tf.keras.losses.MeanSquaredError(
+    reduction=tf.keras.losses.Reduction.SUM_OVER_BATCH_SIZE
+)
+mae = tf.keras.losses.MeanAbsoluteError(
+    reduction=tf.keras.losses.Reduction.SUM_OVER_BATCH_SIZE
+)
+
+"""
+## Loading the Dataset
+
+This example uses the [LJSpeech dataset](https://keithito.com/LJ-Speech-Dataset/).
+
+The LJSpeech dataset is primarily used for text-to-speech and consists of 13,100 discrete
+speech samples taken from 7 non-fiction books, having a total length of approximately 24
+hours. The MelGAN training is only concerned with the audio waves so we process only the
+WAV files and ignore the audio annotations.
+"""
+
+"""shell
+!wget https://data.keithito.com/data/speech/LJSpeech-1.1.tar.bz2
+!tar -xf /content/LJSpeech-1.1.tar.bz2
+"""
+
+# Splitting the dataset into training and testing splits
+wavs = glob.glob("LJSpeech-1.1/wavs/*.wav")
+train, test = train_test_split(wavs, test_size=TEST_SPLIT)
+
+print(f"Number of training audio files: {len(train)}")
+print(f"Number of testing audio files: {len(test)}")
+
+# Mapper function for loading the audio. This function returns two instances of the wave
+
+
+def preprocess(filename):
+    audio = tf.audio.decode_wav(tf.io.read_file(filename), 1, DESIRED_SAMPLES).audio
+    return audio, audio
+
+
+# Create tf.data.Datasets and mapping the dataset
+train_dataset = tf.data.Dataset.from_tensor_slices(train,)
+train_dataset = train_dataset.map(preprocess, tf.data.AUTOTUNE)
+
+test_dataset = tf.data.Dataset.from_tensor_slices(test,)
+test_dataset = test_dataset.map(preprocess, tf.data.AUTOTUNE)
```

Comment:
This dataset is never used

Reply:
Switched to using the full dataset instead